### PR TITLE
feat(dataavailability): EigenDA support

### PR DESCRIPTION
**Pull request by ethenotethan**

### CODE REVIEW

File: ethtxmanager/ethtxmanager.go [hunk 1]
```diff
@@ -70,10 +70,12 @@ func (c *Client) Add(ctx context.Context, owner, id string, from common.Address,
 	}
 	// get gas
 	gas, err := c.etherman.EstimateGas(ctx, from, to, value, data)
+	println("Forced gas: ", c.cfg.ForcedGas)
 	if err != nil {
 		err := fmt.Errorf("failed to estimate gas: %w, data: %v", err, common.Bytes2Hex(data))
 		log.Error(err.Error())
 		if c.cfg.ForcedGas > 0 {
+			println("Using forced gas")
```
Comment (teddyknox) [inline ethtxmanager/ethtxmanager.go]: nit: remove printlines

Comment (ethenotethan) [reply ethtxmanager/ethtxmanager.go]: [570e7ab](https://github.com/Layr-Labs/cdk-validium-node/pull/1/commits/570e7abe8e7a1f50a4835722364ad06e372a5e66)

File: dataavailability/dataavailability.go [hunk 2]
```diff
@@ -113,7 +113,7 @@ func (d *DataAvailability) trustedSequencerData(batchNums []uint64, expectedHash
 		return nil, fmt.Errorf("invalid arguments, len of batch numbers does not equal length of expected hashes: %d != %d",
 			len(batchNums), len(expectedHashes))
 	}
-	nums := make([]*big.Int, 0, len(batchNums))
```
Comment (teddyknox) [inline dataavailability/dataavailability.go]: nit: why this change?

File: dataavailability/eigenda/client.go [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+// ErrInvalidInput is returned when the input is not valid for posting to the DA storage.
+var ErrInvalidInput = errors.New("invalid input")
+
+// DAClient is an HTTP client to communicate with a DA storage service.
+// It creates commitments and retrieves input data + verifies if needed.
+type DAClient struct {
+	url string
+}
+
+func NewDAClient(url string) *DAClient {
```
Comment (teddyknox) [inline dataavailability/eigenda/client.go:24]: Might be nice to define an interface for this so it can be mocked out. Not strictly necessary though.

File: dataavailability/eigenda/eigenda.go [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+	}
+
+	// post the data to EigenDA
+	rawCommit, err := daClient.SetInput(ctx, b)
+	if err != nil {
+		return nil, err
+	}
+
+	// encode version to the rawCommit for posting on-chain
+	return EncodeVersion(rawCommit, 1), nil
```
Comment (teddyknox) [inline dataavailability/eigenda/eigenda.go]: Can we pull this version literal (`1`) into a const at the top of the file?

**Review by teddyknox — APPROVED** [review]
LGTM with one comment

File: dataavailability/eigenda/eigenda.go (hunk 4 above)
Comment (ethenotethan) [reply dataavailability/eigenda/eigenda.go]: [c95ebf6](https://github.com/Layr-Labs/cdk-validium-node/pull/1/commits/c95ebf6fbc366991125f81567ee8c987b440b4f4)

File: dataavailability/dataavailability.go (hunk 2 above)
Comment (ethenotethan) [reply dataavailability/dataavailability.go]: [c95ebf6](https://github.com/Layr-Labs/cdk-validium-node/pull/1/commits/c95ebf6fbc366991125f81567ee8c987b440b4f4)
